src/openms/source/METADATA/SpectrumMetaDataLookup.cpp

```diff
@@ -188,6 +188,34 @@ namespace OpenMS
     return success;
   }
 
+  bool SpectrumMetaDataLookup::addMissingIMToPeptideIDs(vector<PeptideIdentification>& peptides,
+                                const MSExperiment& exp)
+  {
+    SpectrumLookup lookup;
+    bool success = true;
+    lookup.readSpectra(exp.getSpectra());
```

Comment:
@timosachsenberg I cannot find SpectrumLookup::readSpectra is or SpectrumMetaDataLookup:: readSpectra in the source files, but I see it a lot of times in the codebase. Somehow `lookup.empty()` is always true. Input mzML is not empty and loaded as exp in CometAdapter..Any ideas? :)

Reply:
It is in the SpectrumMetaDataLookup.h header file.

    template <typename SpectrumContainer>
    void readSpectra(const SpectrumContainer& spectra, 
                     const String& scan_regexp = default_scan_regexp,
                     bool get_precursor_rt = false)

not sure why it is empty though. Do you get an error like:
Error: Could not extract scan number from spectrum native ID ...

Reply:
yes

Reply:
`Error: Unexpected internal error (the element 'spectrum with native ID 'index=2'' could not be found)`

Reply:
wrong regex pattern?

Reply:
We do not load the spectra in our adapters if not desperately needed.
We only load the metadata of the mzML by default.

Reply:
If you set the regex parameter to "" it will set all scan numbers to -1.
This should not be an issue because lookup is done by native id (and not by scan number).
Not sure why there are no spectra (I thought there should be spectra but they don't contain peaks)

Reply:
metadata_only is different from fill_data

Reply:
Ah right so fill_data needs to be false

Reply:
If the drift time is annotated to the spectrum, it should work.
@jonasscheid do you know why this information is annotated at the spectrum and not at the precursor?
I might be wrong, but as I understood it this is some (e.g. averaged) IM of the precursor ions.

Reply:
I don't know to be honest 🤔 but does not make sense to me, indeed I also thought it is an aggregation of IM values of multiple scans